lib/httplib/httpheaders.go

```diff
@@ -21,51 +21,93 @@ package httplib
 import (
 	"fmt"
 	"net/http"
+	"regexp"
 	"sort"
 	"strings"
+	"sync"
 	"time"
 
-	"golang.org/x/exp/maps"
-
 	"github.com/gravitational/teleport/api/client/proto"
+	"github.com/gravitational/teleport/api/utils"
 )
 
-var defaultContentSecurityPolicy = map[string]string{
-	"default-src": "'self'",
+// Mutex protected cache for memoizing functions which construct the CSP header string.
+// This is necessary because the CSP header is constructed on every request to the web UI,
+// so caching here has a significant performance impact.
+type cspCache struct {
```

Comment:
We could consider `sync.Map` here as it may provide better performance than `RWMutex` (this type of cache is explicitly mentioned in the [docs](https://pkg.go.dev/sync#Map)

Reply:
I tried that here: 53b0d65409b881bc88ee66efe107eecee0a96a62. The thing is, that causes us to need a typecast on every load, the performance of which I'm unsure of. @zmb3 any thoughts?

Reply:
I don't think it matters either way. If you really want to know, the best way to compare the performance of both approaches would be to write a benchmark.